### PR TITLE
Red 973 addto cachelist by serach

### PR DESCRIPTION
**Pull request by Slini11**
### 1. Why is this change necessary?
This change provides the opterunity to add caches by a list to a new or exisiting cachelist.

### 2. What does this change do, exactly?
On the searchpage you can check checkboxes and add those checked caches to a existing cacheliste (dropdown) or new cachelist.

### 4. Please link to the relevant issues (if any).
Please depoly this feature to the testserver first because it needs to be tested!

### 5. Checklist

- [no] I have written tests and verified that they fail without my change
- [ ] I have squashed any insignificant commits
- [ ] This change has comments for package types, values, functions, and non-obvious lines of code

### CODE REVIEW

File: htdocs/search.php [hunk 1]
```diff
@@ -1292,6 +1295,16 @@
             $sql_where[] = '`cache_status`.`allow_user_view`=1';
         }
 
+        //add selected caches to selected cachelist
+        if (isset($_REQUEST['addToList']) && isset($_REQUEST['addCache']) && isset($_REQUEST['selectCachelist'])) {
+            $list_caches_= $_REQUEST['addToList'];
+            $added_waypoints = addToList($list_caches);
+            $addCachelist= cachelist::getListById($_REQUEST['selectCachelist'] + 0); // null for invalid ID
```
Comment (teiling88) [inline htdocs/search.php]: (int) $_REQUEST['selectCachelist'] anstatt + 0 machen bitte.

File: htdocs/search.php [hunk 2]
```diff
@@ -1292,6 +1295,16 @@
...
+        } elseif (isset($_REQUEST['addToList'])) {
+            $addCachelist= cachelist::getListById($_REQUEST['selectCachelist'] + 0); // null for invalid ID
```
Comment (teiling88) [inline htdocs/search.php]: hier bitte auch type cast (int) machen bitte.

File: htdocs/search.php (hunk 1 above)
Comment (Slini11) [reply htdocs/search.php]: Fixed it in new commit :).

File: htdocs/search.php (hunk 2 above)
Comment (Slini11) [reply htdocs/search.php]: Fixed it in new commit :).

**Pull request merged**